pkg/collector/providers/providers.go

```diff
@@ -32,4 +32,6 @@ type ConfigProviderFactory func(cfg config.ConfigurationProviders) (ConfigProvid
 type ConfigProvider interface {
 	Collect() ([]check.Config, error)
 	String() string
+	Watcher()
```

Comment:
Not going into technical details, I understand the reasons why polling might not be the best approach here.
My concerns are at the design level: the config provider was meant to function through polling but we're adding a completely different behaviour here. The fact that we spawn event driven watchers in a function called `pollConfigs` raises few flags. 
If we want to go down the path of refactoring the configuration providers, I'd rethink the entire concept and try to find a clean solution that can accomodate both polling-based and event-driven providers, or get rid entirely of the polling if possible. 
Let's discuss this in a separated design doc?